onnx/test/runtime_test.py

```diff
@@ -0,0 +1,1174 @@
+        X = make_tensor_value_info("X", TensorProto.FLOAT, [None])
+        Y = make_tensor_value_info("Y", TensorProto.FLOAT, [None])
+        node1 = make_node("RandomNormalLike", ["X"], ["Y"], seed=0.0)
+        graph = make_graph([node1], "g", [X], [Y])
+        onnx_model = make_model(graph)
+        check_model(onnx_model)
+        sess = rt.Inference(onnx_model)
+        got = sess.run(None, {"X": np.zeros((2, 4), dtype=np.float32)})[0]
+        self.assertEqual(got.shape, (2, 4))
+        self.assertEqual(got.dtype, np.float32)
+
+    def test_eval_celu(self):
+        inst = Celu.create(alpha=0.5)
+        self.assertEqual(inst.alpha, 0.5)
+        x = np.array([[0, 1], [-1, 2]], dtype=np.float32)
+        y = Celu.eval(x, alpha=0.5)
+        expected = _vcelu1(x, alpha=0.5)
+        assert_almost_equal(expected, y)
+
+    def test_eval_celu_load_op(self):
+        celu = load_op("", "Celu")
+        self.assertEqual(celu.op_domain, "")
+        inst = celu.create(alpha=0.5)
+        self.assertEqual(inst.alpha, 0.5)
+        x = np.array([[0, 1], [-1, 2]], dtype=np.float32)
+        y = celu.eval(x, alpha=0.5)
+        expected = _vcelu1(x, alpha=0.5)
+        assert_almost_equal(expected, y)
+
+    def test_create_adam(self):
+        inst = Adam.create(alpha=0.5)
+        self.assertEqual(inst.alpha, 0.5)
+
+    def test_conv(self):
+        try:
+            import onnxruntime as ort
+        except ImportError:
+            # onnxruntime is not available
+            return
+
+        X = make_tensor_value_info("X", TensorProto.FLOAT, [None, None, None, None])
+        Y = make_tensor_value_info("Y", TensorProto.FLOAT, [None, None, None, None])
+        B = make_tensor_value_info("B", TensorProto.FLOAT, [None, None, None, None])
+        W = make_tensor_value_info("W", TensorProto.FLOAT, [None, None, None, None])
+        node = make_node(
+            "Conv",
+            ["X", "W", "B"],
+            ["Y"],
+            pads=[1, 1, 1, 1],
+            dilations=[1, 1],
+            strides=[2, 2],
+        )
+        graph = make_graph([node], "g", [X, W, B], [Y])
+        onnx_model = make_model(graph, opset_imports=[make_opsetid("", 16)])
+
+        sess1 = ort.InferenceSession(onnx_model.SerializeToString())
+        sess2 = rt.Inference(onnx_model)
+
+        sH, sW = 5, 6
+        for i in range(sH):
+            for j in range(sW):
+                X = np.zeros((1, 1, sH, sW), dtype=np.float32)
+                X[0, 0, i, j] = 1.0
+                W = np.zeros((1, 1, 3, 3), dtype=np.float32)
+                W[0, 0, :, :] = np.minimum(2 ** np.arange(9).reshape((3, -1)), 256)
+
+                B = np.array([[[[0]]]], dtype=np.float32)
+                expected = sess1.run(None, {"X": X, "W": W, "B": B})[0]
+                got = sess2.run(None, {"X": X, "W": W, "B": B})[0]
+                assert_almost_equal(expected, got)
+
+    def test_qlinearconv(self):
+        try:
```

Comment:
We can skip this test if ort is not installed. Example https://github.com/pytorch/pytorch/blob/c7c2578f93fbfad5f769543848642a16b6071756/torch/testing/_internal/common_utils.py#L1204-L1211

Reply:
done